### ISSUE TITLE
opencv-python package not working

### ISSUE THREAD

**Issue by h-ahmad** [HewlettPackard/swarm-learning#105]
#  Issue description
- issue description: import cv2 not working. 
- occurrence - consistent or rare: Containers are started successfully. But training process is not started.
- error messages: 2022-07-18T14:49:43.943250 /tmp/hpe-swarm/demo.1.response.pipe: File exists (Program does not proceed after this line)
# OS and ML Platform
- details of host OS: Ubuntu 20.04
- details of ML platform used: Python3
- details of Swarm learning Cluster (Number of machines, SL nodes, SN nodes): Default (Example of this repository)

# Quick Checklist: Respond [Yes/No]
- APLS server web GUI shows available Licenses? Yes
- If Multiple systems are used, can each system access every other system? Yes
- Is Password-less SSH configuration setup for all the systems? Yes
- If GPU or other protected resources are used, does the account have sufficient privileges to access and use them? Yes
- Is the user id a member of the docker group? Yes

# Additional notes
- Are you running documented example without any modification? No
- Add any additional information about use case or any notes which supports for issue investigation: Other code works normally, but the only issue is with the line 'import cv2'. Training process is successful without this line.

**Comment by IMOKURI** [comment]
I think you need to add commands to container image build task that install `opencv-python` dependencies.
For example,

```diff
--- mnist/swci/taskdefs/user_env_tf_build_task.yaml
+++ mnist/swci/taskdefs/user_env_tf_build_task.yaml
@@ -11,6 +11,7 @@ Body:
   BuildType: INLINE
   BuildSteps:
     - FROM tensorflow/tensorflow:2.7.0
+    - RUN apt-get update && apt-get install -y --no-install-recommends libopencv-dev
     - " "
     - RUN pip3 install --upgrade pip && pip3 install \
     - "   keras matplotlib opencv-python pandas protobuf==3.15.6 sklearn"
```

Note: You need to delete built image before rerun task.
Something like this;

```
docker rmi user-env-tf2.7.0-swop:latest
```


I think this message can be ignored for this issue.

> `/tmp/hpe-swarm/demo.1.response.pipe: File exists`

**Comment by h-ahmad** [comment]
You are right, but there is already a command to install opencv-python package as attached.
![image](https://user-images.githubusercontent.com/28649744/179684066-141c472f-0eb7-4995-a341-0b275b1366e1.png)
Moreover, I always kill/stop all containers before restarting the whole process.

**Comment by IMOKURI** [comment]
Depending on the Docker base image, `opencv-python` cannot be used with just a pip installation.

You need to install the dependent packages via OS package manager.

So I think this example is not good to use opencv-python...

For example, if you run the built image with `docker run`, `import cv2` will fail with an insufficient library error.


![image](https://user-images.githubusercontent.com/1638500/179687833-a4097629-43e2-4ce2-a90b-b9cdb83cfd2b.png)

**Comment by h-ahmad** [comment]
> 

Thanks for your time and valuable answer. Can you please explain or mentioned the commands what should be added or removed in the file mentioned above? I want to run this package in the docker container. Thanks.

**Comment by IMOKURI** [comment]
It may be too late, but with the following update I was able to use `import cv2`.

```diff
@@ -12,9 +12,10 @@ Body:
   BuildSteps:
     - FROM pytorch/pytorch:1.11.0-cuda11.3-cudnn8-runtime
     - " "
+    - ENV DEBIAN_FRONTEND noninteractive
     - RUN apt-get update && apt-get install           \
     - '   build-essential python3-dev python3-pip     \'
-    - "   python3-setuptools --no-install-recommends -y"
+    - "   python3-setuptools libopencv-dev --no-install-recommends -y"
     - " "
     - RUN conda install pip
     - " "

```
